numba/tests/test_parfors.py

```diff
@@ -497,6 +497,58 @@ def test_impl(x, is_positive, N):
         is_positive = np.zeros(N)
         self.check(test_impl, x, is_positive, N)
 
+    @skip_unsupported
+    def test_reduce(self):
+        def test_impl(A):
+            init_val = -1
+            return reduce(lambda a,b: min(a, b), A, init_val)
+
+        A = np.random.ranf(4)
+        cpfunc = self.compile_parallel(test_impl, (numba.typeof(A),))
+        self.assertEqual(cpfunc.entry_point(A), test_impl(A))
+        A = np.random.randint(10, size=4).astype(np.int32)
+        cpfunc = self.compile_parallel(test_impl, (numba.typeof(A),))
+        self.assertEqual(cpfunc.entry_point(A), test_impl(A))
+
+    @skip_unsupported
+    def test_min(self):
+        def test_impl(A):
+            return A.min()
+
+        A = np.random.ranf(4)
+        self.check(test_impl, A)
+        A = np.random.randint(10, size=4).astype(np.int32)
+        self.check(test_impl, A)
+
+    @skip_unsupported
+    def test_max(self):
+        def test_impl(A):
+            return A.max()
```

Comment:
I assume the np versions before canonicalisation has taken place also work (`np.max(A)`)? Perhaps they should also be tested?

Reply:
Yes, they work. I added the tests.